src/domain/useCase/sslCertificateWatchDog.go

```diff
@@ -0,0 +1,129 @@
+	sslCmdRepo repository.SslCmdRepo,
+	vhostQueryRepo repository.VirtualHostQueryRepo,
+	vhostCmdRepo repository.VirtualHostCmdRepo,
+) SslCertificateWatchDog {
+	return SslCertificateWatchDog{
+		sslQueryRepo:   sslQueryRepo,
+		sslCmdRepo:     sslCmdRepo,
+		vhostQueryRepo: vhostQueryRepo,
+		vhostCmdRepo:   vhostCmdRepo,
+	}
+}
+
+func (uc SslCertificateWatchDog) createInlineHtmlMapping(
+	vhost valueObject.Fqdn,
+	ownershipHash string,
+) error {
+	path, _ := valueObject.NewMappingPath(OwnershipValidatePath)
+	matchPattern, _ := valueObject.NewMappingMatchPattern("equals")
+	targetType, _ := valueObject.NewMappingTargetType("inline-html")
+	httpResponseCode, _ := valueObject.NewHttpResponseCode(200)
+	inlineHtmlContent, _ := valueObject.NewInlineHtmlContent(ownershipHash)
+
+	inlineHmtlMapping := dto.NewCreateMapping(
+		vhost,
+		path,
+		matchPattern,
+		targetType,
+		nil,
+		nil,
+		&httpResponseCode,
+		&inlineHtmlContent,
+	)
+	return uc.vhostCmdRepo.CreateMapping(inlineHmtlMapping)
+}
+
+func (uc SslCertificateWatchDog) Execute() {
+	vhosts, err := uc.vhostQueryRepo.Get()
+	if err != nil {
+		log.Printf("FailedToGetVhosts: %s", err.Error())
+		return
+	}
+
+	invalidSslVhosts := []valueObject.Fqdn{}
+	for _, vhost := range vhosts {
+		isSslValid := uc.sslQueryRepo.IsSslPairValid(vhost.Hostname)
+		if isSslValid {
+			continue
+		}
+
+		invalidSslVhosts = append(invalidSslVhosts, vhost.Hostname)
+	}
+
+	for _, invalidSslVhost := range invalidSslVhosts {
+		sslPair, err := uc.sslQueryRepo.GetSslPairByHostname(invalidSslVhost)
+		if err != nil {
+			log.Printf("FailedToGetSslPair (%s): %s", invalidSslVhost, err.Error())
+			continue
+		}
+
+		ownershipHash := uc.sslQueryRepo.GetOwnershipHash(sslPair.Certificate.CertificateContent)
+
+		err = uc.createInlineHtmlMapping(invalidSslVhost, ownershipHash)
+		if err != nil {
+			log.Printf("FailedToCreateOwnershipValidationMapping: %s", err.Error())
+			continue
+		}
+
+		// Wait for NGINX reload
+		time.Sleep(5 * time.Second)
+
+		isOwnershipValid := uc.vhostQueryRepo.IsDomainOwner(invalidSslVhost, ownershipHash)
+		if !isOwnershipValid {
+			log.Printf("CurrentHostIsNotDomainOwner: %s", invalidSslVhost.String())
+		}
+
+		vhostMappings, err := uc.vhostQueryRepo.GetMappingsByHostname(invalidSslVhost)
+		if err != nil {
+			log.Printf("FailedToGetVhostMappings: %s", err.Error())
+			continue
+		}
+
+		if len(vhostMappings) == 0 {
+			log.Printf("VhostMappingsNotFound: %s", invalidSslVhost)
+			continue
+		}
+
+		lastMappingIndex := len(vhostMappings) - 1
+		lastMapping := vhostMappings[lastMappingIndex]
+		err = uc.vhostCmdRepo.DeleteMapping(lastMapping)
+		if err != nil {
+			log.Printf("FailedToDeleteOwnershipValidationMapping: %s", err.Error())
+			continue
+		}
+
+		if !isOwnershipValid {
+			continue
+		}
+
+		err = uc.sslCmdRepo.ReplaceWithValidSsl(invalidSslVhost)
+		if err != nil {
+			log.Printf("FailedToReplaceWithValidSsl: %s", err.Error())
+		}
+	}
```

Comment:
Nada disso aqui é regra de negócio. Pelo que entendi, ownershipHash é a hash de validação. Então o nome certo seria ownershipValidationHash e com certeza não deveria estar no UC. Isso é claramente lógica de infra, não tem negócio algum aqui.

Única regra de negócio do seu UC é filtrar quem é ou não é válido como eu expliquei. Daí você chamaria o ReplaceWithValidSsl num for e toda essa regra de criar o ownership validation hash e tals ficaria na infra. O ReplaceWithValidSsl retornaria os erros possíveis, tais como domínio não apontando para o cloud ou outros possíveis erros.